composer.json

```diff
@@ -22,6 +22,7 @@
         "php-http/message": "^1.2",
         "psr/http-client": "^1.0",
         "psr/http-factory": "^1.0",
+        "psr/http-factory-implementation": "^1.0",
```

Comment:
or should we be opinionated and depend on a concrete implementation, so that this client is easy to install?

if we add this, people need to chose one of the factory implementations which makes installation more complicated.

Reply:
discovery will auto-install a factory if none is selected.